keras/src/backend/tensorflow/numpy.py

```diff
@@ -2382,6 +2382,12 @@ def _negative_k_branch():
         mask = i >= j - k
         return tf.where(tf.broadcast_to(mask, shape), x, tf.zeros_like(x))
 
+    if isinstance(k, int):
+        if k >= 0:
+            return tf.linalg.band_part(x, -1, k)
+        return _negative_k_branch()
+
+    # when `k` is a tensor
     return tf.cond(
         k >= 0, lambda: tf.linalg.band_part(x, -1, k), _negative_k_branch
```

Comment:
Actually, we turn off autograph, so this should be: `tf.greater_equal(k, 0)` instead of `k >= 0`.

I think it works with the `__ge__` override, but I would prefer to be explicit.